app/models.py

```diff
@@ -0,0 +1,106 @@
+from flask_sqlalchemy import SQLAlchemy
+from flask_migrate import Migrate
+from sqlalchemy.ext.declarative import declared_attr
+import sqlalchemy as sa
+
+migrate = Migrate()
+
+
+class IdModel(object):
+    @declared_attr
+    def id(cls):
+        for base in cls.__mro__[1:-1]:
+            if getattr(base, "__table__", None) is not None:
+                type = sa.ForeignKey(base.id)
+                break
+        else:
+            type = sa.Integer
+
+        return sa.Column(type, primary_key=True)
+
+
+db = SQLAlchemy(model_class=IdModel)
+
+
+class NameDescriptionMixin(object):
+    name = db.Column(db.String(128))
+    description = db.Column(db.Text())
+
+
+class SkillsMixin(object):
```

Comment:
How will this be used?

Reply:
Good catch - I think I had a very over-complicated plan for this but since I don't currently use it I'll bin it and bring it back if necessary